lab_stack/ConsoleApplication33/ConsoleApplication33/Stack.cpp

```diff
@@ -30,25 +29,25 @@
                 previous = current;
                 current = current->next;
             }
-            if (current == nullptr) //��������� ������� � �����
+            if (current == nullptr) //ïîñëåäíèé ýëåìåíò â ñòåêå
             {
                 previous->next = element_stack;
             }
-            else //������ ������� � ������� �����������
+            else //íàéäåí ýëåìåíò ñ ìåíüøèì ïðèîðèòåòîì
             {
                 previous->next = element_stack;
                 element_stack->next = current;
             }
         }
     }
 
-    //������������ �������� �������� �� �����
+    //âûòàëêèâàíèå âåðõíåãî ýëåìåíòà èç ñòåêà
     void Stack::Pop()
     {
         Element value = *top;
         if (IsEmpty())
         {
-            std::cout << "\x1b[31m���� ����!\x1b[0m" << std::endl;
+            std::cout << "\x1b[31mÑòåê ïóñò!\x1b[0m" << std::endl;
         }
         else
         {

```

Comment:
You don`t need destructor here. You need only do something with next element (example: if we poped 1 element, top-1 element will point to nullptr)

Reply:
I understood. I will not call destructor myself.